internal/store.go

```diff
@@ -1081,6 +1104,10 @@ func (s *LoadingStore[K, V]) Get(ctx context.Context, key K) (V, error) {
 		loaded, err, _ := shard.group.Do(key, func() (Loaded[V], error) {
 			// load and store should be atomic
 			shard.mu.Lock()
+			if shard.closed {
+				shard.mu.Unlock()
```

Comment:
I checked out the code locally and I see that the locking & unlocking code gets weird, in that sometimes a function locks but doesn't unlock (hopefully because it calls a function that does the unlock). You risk introducing deadlock scenarios with this. 

My recommendation (not for this PR) is that you at least document each function:

- if it unlocks a mutex
- if it assumes that the mutex has already been locked by the caller.

I also recommend that you use this pattern:

```
mutex.Lock()
defer mutex.Unlock()
```

to prevent the issue completely (at the expense that you will be holding the mutex for a longer time).